write-fonts/src/tables/glyf.rs

```diff
@@ -86,29 +88,291 @@ pub enum BadKurbo {
     HasCubic,
     TooSmall,
     MissingMove,
+    UnequalNumberOfElements(Vec<usize>),
+    InconsistentPathElements(usize, Vec<&'static str>),
+}
+
+/// Point with an associated on-curve flag.
+///
+/// Similar to read_fonts::tables::glyf::CurvePoint, but uses kurbo::Point directly
+/// thus it does not require (x, y) coordinates to be rounded to integers.
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct ContourPoint {
+    point: kurbo::Point,
+    on_curve: bool,
+}
+
+impl ContourPoint {
+    fn new(point: kurbo::Point, on_curve: bool) -> Self {
+        Self { point, on_curve }
+    }
+
+    fn on_curve(point: kurbo::Point) -> Self {
+        Self::new(point, true)
+    }
+
+    fn off_curve(point: kurbo::Point) -> Self {
+        Self::new(point, false)
+    }
+
+    fn distance(&self, other: &Self) -> f64 {
+        self.point.distance(other.point)
+    }
+}
+
+impl From<ContourPoint> for CurvePoint {
+    fn from(pt: ContourPoint) -> Self {
+        let (x, y) = pt.point.ot_round();
+        CurvePoint::new(x, y, pt.on_curve)
+    }
+}
+
+/// A helper struct for building interpolatable contours
+///
+/// Holds a vec of contour, one contour per glyph.
+#[derive(Clone, Debug, PartialEq)]
+struct InterpolatableContourBuilder(Vec<Vec<ContourPoint>>);
+
+impl InterpolatableContourBuilder {
+    /// Create new set of interpolatable contours beginning at the provided points
+    fn new(move_pts: Vec<kurbo::Point>) -> Self {
```

Comment:
well, it's doing into_iter internally so I believe it is consuming the vecs. Note this whole struct is private, it doesn't need to be nice. These points are not usually already in a continuous slice, the caller has to make a vector anyway since the points belong to distinct BezPaths. Maybe I'm missing something else..

Reply:
the "reused" part is what I don't quite understand at the moment (by what, where?) but I will try to do what you say and hopefully rustc will guide me through it ;)